lib/chef/provider/remote_file/cache_control_data.rb

```diff
@@ -153,7 +153,15 @@ def sanitized_cache_file_basename
           # human-readable but within the bounds of local file system
           # path length limits
           scrubbed_uri = uri.gsub(/\W/, '_')[0..63]
-          uri_md5 = Chef::Digester.instance.generate_md5_checksum(StringIO.new(uri))
+          uri_md5 = if Chef::Config.fips_mode
```

Comment:
Yeah, do it for everyone and transparently upgrade from MD5 to SHA2 on reads without re-download.